Parse build output of GitHub Actions

hakonsbm opened the pull request:
GitHub actions will now parse the output of the build, give a proper summary, and fail if there are new compiler warnings raised, like it used to with Travis. This fixes #2005.

Also, the `parse_travis_log.py` script is renamed to `parse_build_log.py`, and trailing whitespace is cleaned up in the modified files.

heplesser reviewed (commented):
@hakonsbm Thanks, this looks good. But I noticed that for the logs I checked, it shows "0 warnings (0 expected)" in all cases, including FULL. Don't we have some cases with a small number of warnings? Or are they so thoroughly handled by the parser that they do not count as expected?

I also noticed that in the summary at the end of build and tests it still says "Travis" in the header. Could you fix this here?

hakonsbm commented:
@heplesser 
> I noticed that for the logs I checked, it shows "0 warnings (0 expected)" in all cases, including FULL. Don't we have some cases with a small number of warnings? Or are they so thoroughly handled by the parser that they do not count as expected?

Most of that was fixed in #1852, and the rest in #1830. So, except for a few warnings from SLI that are handled by the parser, we are expecting zero warnings in all cases. I've removed it from the summary.

> I also noticed that in the summary at the end of build and tests it still says "Travis" in the header. Could you fix this here?

Fixed!

gtrensch reviewed (approved):
Thanks for this contribution. Looks good to me :+1:.
